src/traces/pie/defaults.js

```diff
@@ -70,6 +70,10 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         if(hasOutside) {
             coerce('automargin');
         }
+
+        if(textposition !== 'none' && textposition !== 'outside') {
```

Comment:
Wouldn't `textposition === 'inside' || textposition === 'auto'` be more readable?

Reply:
Good point.

Reply:
Actually that does not work; whereas `textposition` could be also an array.

Reply:
Ok well, I'd vote for:

```js
if(textposition === 'inside' || textposition === 'auto || Array.isArray(textposition)) {
  // ...
}
```

Reply:
Fixed in https://github.com/plotly/plotly.js/pull/4420/commits/a27dc29ca87303fd53bf994bfe1203b3c5e96887.